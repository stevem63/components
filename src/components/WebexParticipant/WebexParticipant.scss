@import '../../styles/variables';

.#{$WEBEX_COMPONENTS_CLASS_PREFIX}-participant {
  display: flex;
  padding: 0.625rem 0 0.625rem 0;

  .participant-name {
    @include body-fonts;
    align-items: center;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    padding-left: 0.25rem;
  }
}
