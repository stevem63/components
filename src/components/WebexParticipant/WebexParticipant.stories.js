import React from 'react';
import {storiesOf} from '@storybook/react';

import jsonData from '../../data';
import {WebexJSONAdapter} from '../../adapters';

import WebexParticipant from './WebexParticipant';

import {WebexDataProvider} from '..';

// Setup for the stories
const stories = storiesOf('Webex Participant', module);
const webexAdapter = new WebexJSONAdapter(jsonData);

// Stories
stories.add('default', () => (
  <WebexDataProvider adapter={webexAdapter}>
    <WebexParticipant personID="default" />
  </WebexDataProvider>
));
