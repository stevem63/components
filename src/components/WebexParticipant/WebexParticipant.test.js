import React from 'react';

import WebexParticipant from './WebexParticipant';

jest.mock('../hooks/usePerson');

describe('Webex Participant component', () => {
  describe('snapshot', () => {
    test('matches snapshot of webex participant', () => {
      expect(shallow(<WebexParticipant personID="default" />)).toMatchSnapshot();
    });
  });
});
