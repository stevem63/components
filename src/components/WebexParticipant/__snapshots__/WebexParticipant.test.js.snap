// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webex Participant component snapshot matches snapshot of webex participant 1`] = `
<ContextTransform(ListItem)
  className="wxc-participant"
>
  <WebexAvatar
    displayStatus={false}
    personID="default"
  />
  <div
    className="participant-name"
  >
    Webex Component User
  </div>
</ContextTransform(ListItem)>
`;
